src/tools/ipv6-ula-generator/ipv6-ula-generator.vue

```diff
@@ -0,0 +1,64 @@
+<template>
+  <div>
+    <n-space vertical :size="50">
+      <n-alert title="Info" type="info">
+        This page uses the first method suggested by IETF using the current timestamp plus the mac address, sha1 hashed,
+        and the lower 40 bits to generate your random ULA.
+      </n-alert>
+
+      <n-form-item label="MAC address:" v-bind="validationAttrs">
+        <n-input
+          v-model:value="macAddress"
+          size="large"
+          placeholder="Type a MAC address"
+          clearable
+          autocomplete="off"
+          autocorrect="off"
+          autocapitalize="off"
+          spellcheck="false"
+        />
+      </n-form-item>
+    </n-space>
+
+    <div v-if="validationAttrs.validationStatus !== 'error'">
+      <n-input-group v-for="{ label, value } in calculatedSections" :key="label" style="margin: 5px 0">
+        <n-input-group-label style="flex: 0 0 120px"> {{ label }} </n-input-group-label>
+        <input-copyable :value="value" readonly />
+      </n-input-group>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { SHA1 } from 'crypto-js';
+import InputCopyable from '@/components/InputCopyable.vue';
+import { macAddressValidation } from '@/utils/macAddress';
+
+const macAddress = ref('20:37:06:12:34:56');
+const calculatedSections = computed(() => {
+  const timestamp = new Date().getTime();
+  const hex40bit = SHA1(timestamp + macAddress.value)
+    .toString()
```

Comment:
Maybe use [`useTimestamp`](https://vueuse.org/core/useTimestamp/#usetimestamp) for reactive live computation ?
Don't no if it's a good idea

```suggestion

const timestamp = useTimestamp({ offset: 0 })
const macAddress = ref('20:37:06:12:34:56');
const calculatedSections = computed(() => {
  const hex40bit = SHA1(timestamp.value + macAddress.value)
    .toString()
```

Reply:
I don't think that this is senseful. As the timestamp is used (in addition to the mac address) to build the sha1 the resulting ula would change every millisecond on the page. This would be very confusing I think. ;)